### DISCUSSION TITLE
Do Perpective Support 100 million records on UI

### DISCUSSION THREAD

**Discussion by akash-goel** [perspective-dev/perspective#1999] [Q&A] [answered]
Hi Team,

we are working on big data and want to check that can we render 100 million records on UI using Perpective ? we have tried with loading 1 million on UI using Perspective and getting error when performing some action on it .

Regards,
Akash

**Comment by timkpaine** [comment, marked as answer]
You can load as much data as your browser has room for in memory, but at a certain point it makes much more sense to [virtualize your data behind a server](https://perspective.finos.org/docs/server#server-only)
